### ISSUE TITLE
Multiple tokens per registry in auth.json

### ISSUE THREAD

**Issue by lcarva** [containers/image#981]
The [auth.json](https://github.com/containers/image/blob/fe9e1741e53189457e49f8014b8d4c79f4fc32fe/docs/containers-auth.json.5.md) file supports a single entry for any given registry. In some cases, this is not sufficient. For example, robot accounts in quay.io are limited in scope by an organization. To access private repositories across two different organizations, two different robot tokens are required.

Would it be possible to enhance the syntax of auth.json to allow multiple authentication tokens for the same registry, but at different paths? For example:
```json
{
	"auths": {
		"quay.io": {
			"auth": "erfi7sYi89234xJUqaqxgmzcnQ2rRFWM5aJX0EC="
		},
		"quay.io/spam": {
			"auth": "juQAqGmz5eR1ipzx8Evn6KGdw8fEa1w5MWczmgY="
		},
		"quay.io/ham": {
			"auth": "Zm9vYmFyOnN1cGVyc2VjcmV0"
		}
	}
}
```

**Comment by lcarva** [comment]
ok. This appears to work :smile: 

Is the syntax documented somewhere?

**Comment by mtrmac** [comment]
I’m afraid there is a very long-standing implementation precedent (https://github.com/moby/moby/blob/f681590a2533829b88850b7e605a000cb538f102/registry/auth.go#L204 ) for that syntax to be interpreted as `quay.io`, ignoring any suffix.

Right now, at the library level, you can use different `SystemContext` values for every image reference, either containing the credentials directly or pointing at separate files.

**Comment by mtrmac** [comment]
This was implemented in #1278 (and updates to that code).
